fastlane/actions/check_bump_type.rb

```diff
@@ -0,0 +1,66 @@
+module Fastlane
+  module Actions
+    class CheckBumpTypeAction < Action
+
+      # Get the type of release desired
+      # If there's only one bump_type allowed it will use that one
+      # If there isn't a bump_type specified and there are many allowed bump_types, it will ask the user
+
+      FIRST_VERSION = "0.0.0".freeze
+      ALL_BUMP_TYPES = %i(build patch minor major).freeze
+      BUILD_CONFIGURATION_ALLOWED_BUMP_TYPES = {
+        test: [],
+        qa: [:build],
+        testflight: ALL_BUMP_TYPES,
+        release: ALL_BUMP_TYPES 
+      }.freeze
+
+      def self.run(params)
+        is_first_deploy = params[:version] == FIRST_VERSION
+        allowed_bump_types = BUILD_CONFIGURATION_ALLOWED_BUMP_TYPES[params[:build_configuration]]
+
+        if is_first_deploy
+          return :major
+        end
+
+        if allowed_bump_types.count == 1
+          return allowed_bump_types.first # The only allowed bump type possible
+        end
+
+        bump_type = params[:bump_type]
+        while not allowed_bump_types.include? bump_type do
+          bump_type = UI.input "Choose the `bump_type` representing the type of deploy. It can be any of %s" % allowed_bump_types.to_s
+        end
+
+        if bump_type != :major
+          UI.important "This seems to be your first deploy. It must be done as `%s`." % :major
+          if (UI.input "Do you want to make it as `%s`? y/N" % :major) == "y"
+            bump_type = allowed_bump_types.last
```

Comment:
Why do we even ask this if we erroring later? In case the deploy HAS to be done like 0.0.1 for example, so you have to do it by hand?

Reply:
The first deploy should always be 1.0.0. That doesn't mean you're going to send that version to the appstore, it just means you're sending a build for version 1.0.0 to testflight. You can iterate that first version with as meny builds as you want but eventually when sending it to review it'll be version 1.0.0.

As for why do we ask for it, I guess it's just to let the user know that it'll bump the major number and not doing it without letting him know. @pablog am I right?

Reply:
Yes, the version numbers used to deploy apps in test flight doesn't need to match the one that will be eventually sent to review.

We just agreed to start in 1.0.0.

Reply:
Great! (Then I'd just don't ask and directly say something like "Do you want to continue?", only because a little repetitive and we are actually not giving another choice.)